📖 Improve Readme (#445)
process-engine/process_engine_runtime: README.md

Please make sure that you run the shell you use for
the installation as **Administrator**.

Also, each full releae provides ready-to-use source files for each platform.
These are stored in a `.tar.gz` archive (for macOS and Linux) and a zip file (for windows).

> All these source have been pre installed and build. You only need to download and upack them

Reviewer: ```suggestion
All these sources have been pre-installed and build. You only need to download and unpack them
```

and you are good to go.

## How do I use this project?

### Starting the ProcessEngineRuntime

Start the application on your main machine with the following command:

```bash
process-engine
```

When started, the ProcessEngine is available at

`http://localhost:8000`.

__Note:__ If you're on Windows and the command `process-engine` can not be
found, please make sure your `PATH` is set correctly.

### Global routes

> The ProcessEngine exposes a number of global http routes,

Reviewer: ```suggestion
The ProcessEngine exposes a number of global HTTP routes,
```

which you can use to get general information about the application.

These routes include:

- `http://localhost:8000/` - Base route to get basic details about the ProcessEngine
- `http://localhost:8000/process_engine` - Same as above
- `http://localhost:8000/security/authority` - Returns the address of the authority
  the ProcessEngine uses to perform claim checks
- `http://localhost:8000/process_engine/security/authority` - Same as above

You might wonder why we use two routes for each UseCase.

The reason is simple:

> Lets say you want to embedd your ProcessEngine into another web application.

Reviewer 2: ```suggestion
Lets say you want to embed your ProcessEngine into another web application.
```

> Lets say you want to embed your ProcessEngine into another web application.

Reviewer 2: ```suggestion
Let's say you want to embed your ProcessEngine into another web application.
```

Usually, you'd want to use routes like `http://localhost:8000/` for your own
purposes and not have it expose information about any embedded service
(which is what the ProcessEngine would be in this instance).

> BPMN Studio uses these global routes to identity remote ProcessEngines to connect to.

Reviewer: ```suggestion
BPMN Studio uses these global routes to identify remote ProcessEngines to connect to.
```

> The route `http://localhost:8000/process_engine` ensures that the Studio can do so, even if

Reviewer: ```suggestion
The route `http://localhost:8000/process_engine` ensures that the studio can do so, even if
```

`http://localhost:8000/` is reserved by your application.

> In other words: These routes allow you to access an embedded ProcessEngine through BPMN studio.

Reviewer: ```suggestion
In other words: These routes allow you to access an embedded ProcessEngine through BPMN Studio.
```

**Note:**
See the [Other startup parameters](#other_startup_parameters) section for instructions
on how to prevent the ProcessEngine from using these global routes.

#### Switching the database

By default, the ProcessEngine will use `SQLite` as its database.

The corresponding files will be placed in the `databases` directory mentioned in the
[Application Files](#application_files) section.

If you want to use a different database, you must provide a `NODE_ENV` parameter at startup:

```bash
NODE_ENV=postgres process-engine
```

Currently supported values are `postgres` and `mysql`.

Each environment comes with its own config.

See:

- [Configuration for mysql repositories](./config/mysql/process_engine)
- [Configuration for postgres repositories](./config/postgres/process_engine)
- [Configuration for sqlite repositories](./config/sqlite/process_engine)

**Note:**

> Switching to mysql or postgres requires an instance of the respective database to be running and accessible!s

Reviewer: ```suggestion
Switching to MySQL or Postgres requires an instance of the respective database to be running and accessible!
```

and the environment you wish to use is named `production`.

Your configs must then be located in the following path:

- macOS: `/Users/{{YOUR_USERNAME}}/runtime/production`
- Linux: `/home/{{YOUR_USERNAME}}/runtime/production`
- Windows: `C:\Users\{{YOUR_USERNAME}}\runtime\production`

You would need to provide the following environment parameters to access this config:

- `NODE_ENV`: `production`
- `CONFIG_PATH`:
    - macOS: `/Users/{{YOUR_USERNAME}}/runtime`
    - Linux: `/home/{{YOUR_USERNAME}}/runtime`
    - Windows: `C:\Users\{{YOUR_USERNAME}}\runtime`

The full start command will then look like this:

- macOS: `CONFIG_PATH=/Users/{{YOUR_USERNAME}}/runtime NODE_ENV=production process-engine`
- Linux: `CONFIG_PATH=/home/{{YOUR_USERNAME}}/runtime NODE_ENV=production process-engine`
- Windows: `CONFIG_PATH=C:\Users\{{YOUR_USERNAME}}\runtime NODE_ENV=production process-engine`

#### Other startup parameters

There are various other parameters you can provide at startup:

> - `NO_HTTP`: Providing this flag will disable all http endpoints of the ProcessEngine.

Reviewer: ```suggestion
- `NO_HTTP`: Providing this flag will disable all HTTP endpoints of the ProcessEngine.
```

>   This can be useful, if you are embedding the ProcessEngine into another application

Reviewer: ```suggestion
  This can be useful if you are embedding the ProcessEngine into another application
```

>   and you do not intend for the ProcessEngine to expose its own http routes

Reviewer: ```suggestion
  and you do not intend for the ProcessEngine to expose its own HTTP routes
```
